backend/tests/hmc-course.test.ts

```diff
@@ -2,10 +2,46 @@ import { describe, test, expect } from "@jest/globals";
 import type * as APIv4 from "hyperschedule-shared/api/v4";
 
 import { linkCourseData, extractSectionTerm } from "../src/hmc-api/course";
+import { parseCourseBoomi } from "../src/hmc-api/files";
 import * as fs from "fs/promises";
 
 const PARSED_SAMPLE_PATH = "src/hmc-api/sample/parsed-sample-v4.json";
 
+const BOOMI_DB_SAMPLE = `
+DBSTART|85ba3f56-d039-43dd-ad22-913c0e2b42b8|2|@|BEGIN|2|@|OUT_START|3|@|SMPL010A EX|^|Example Class|^|SAFR|^||^|SC|^|Example Description
+|#|SMPL127  EX|^|Another Example Class|^|SAFR|^|05.0201|^|PZ|^|Indeed, another example description
+|#|SMPL123  EX|^|Class With Not enough fields|^|PZ|^|this one should be skipped
+|#|SMPL124  EX|^|Class With too many fields|^|SAFR|^||^|PZ|^|the next field should be ignored|^|junk|^||^|
+|#|SMPL191  EX|^|Example Class Boogaloo|^|PAFR|^||^|PO|^|Yeehaw|#||#|OUT_END|3|@|END|2|@|DBEND|85ba3f56-d039-43dd-ad22-913c0e2b42b8|2|@|
+`;
+const BOOMI_DB_EXPECTED: ReturnType<typeof parseCourseBoomi> = [
```

Comment:
would be nice to use previous suggestion of exporting the `BoomiCourse` type and referencing that instead (more explicit/direct) instead of relying on `ReturnType` inference here.